web/src/lib/components/Button.svelte

```diff
@@ -0,0 +1,35 @@
+<script lang="ts">
+	export let onClick: () => void = () => {};
+	export let type: 'button' | 'reset' | 'submit' = 'button';
+	export let isDisabled: boolean = false;
+	let className: string = "";
+	export { className as class };
+</script>
+
+<button class={className} {type} on:click={onClick} disabled={isDisabled}>
+	<slot></slot>
+</button>
```

Comment:
First off, the `{type}` is just a shorthand for `type={type}`. Just a little syntax sugar Svelte offers cause it's pretty common for this to happen: surfacing the attributes of the main component attributes up so that the component itself (not the main component) sort of has those attributes.

And yeah, the exports are like function signatures if you think of components are just functions (though a component would have more in their signature than just what's exported).

The docs about `type` is https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#type. It's optional and has no default value, but I found that it's generally useful to default it to `button` cause it's most common for us to add event listeners to them.